storage/interface.go

```diff
@@ -34,7 +34,7 @@ var (
 // Appendable allows creating appenders.
 type Appendable interface {
 	// Appender returns a new appender for the storage.
-	Appender() Appender
+	Appender(ctx context.Context) Appender
```

Comment:
Can we write a comment here that we might use the context to pass in more information or for tracing but not for timeouts?

Reply:
I think that this should be the choice of the implementation. Maybe we can add that the appender might decide not to use the context or check for context errors.

Reply:
>  appender might decide not to use the context or check for context errors.

This sounds better and more general.

Reply:
Added comment.